### DISCUSSION TITLE
Flux upgrade from v.14.0 to v.31.x

### DISCUSSION THREAD

**Discussion by slimshetty** [fluxcd/flux2#2870] [Q&A] [answered]
We have multiple dev and prod k8 clusters deployed using flux. All the systems now are currently using flux v.14.0. We would like to upgrade flux to .31.x . It is not possible to upgrade all envs at once. 
Once we start upgrading, some envs would be on the latest version whereas some would be on the older version, **is it possible to install multiple versions of the flux cli** on a system so that users can run commands against different versions of flux. 
We also have a central machine which runs flux commands against multiple envs during deployment , how could this machine interact with old and new versions of flux??

**Comment by stefanprodan** [comment, marked as answer]
Yes you can download the Flux CLI binaries from [GitHub Releases](https://github.com/fluxcd/flux2/releases) into different directories, then run the version you need from disk, e.g. `/tmp/flux-0.14/flux <command>` and `/tmp/flux-0.31/flux <command>`. Same in your CI, add the Flux version as a CI parameter, have the CI download that version of the CLI and use it.

**Reply by slimshetty** [reply]
Thanks for the response, yes we will follow this approach during our upgrade
